packages/schematics/angular/migrations/update-15/remove-platform-server-exports_spec.ts

```diff
@@ -0,0 +1,90 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import { EmptyTree } from '@angular-devkit/schematics';
+import { SchematicTestRunner, UnitTestTree } from '@angular-devkit/schematics/testing';
+
+describe('Migration to delete platform-server exports', () => {
+  const schematicName = 'remove-platform-server-exports';
+
+  const schematicRunner = new SchematicTestRunner(
+    'migrations',
+    require.resolve('../migration-collection.json'),
+  );
+
+  let tree: UnitTestTree;
+
+  beforeEach(() => {
+    tree = new UnitTestTree(new EmptyTree());
+  });
+
+  const testTypeScriptFilePath = './test.ts';
+
+  describe('Migration to import() style lazy routes', () => {
+    beforeEach(async () => {
+      tree = new UnitTestTree(new EmptyTree());
+      tree.create('/package.json', JSON.stringify({}));
+    });
+
+    it(`should delete '@angular/platform-server' export when 'renderModule' is the only exported symbol`, async () => {
+      tree.create(
+        testTypeScriptFilePath,
+        `
+          import { Path, join } from '@angular-devkit/core';
+          export { renderModule } from '@angular/platform-server';
+        `,
+      );
+
+      const newTree = await schematicRunner.runSchematicAsync(schematicName, {}, tree).toPromise();
+      const content = newTree.readContent(testTypeScriptFilePath);
+      expect(content).not.toContain('@angular/platform-server');
+      expect(content).toContain(`import { Path, join } from '@angular-devkit/core';`);
+    });
+
+    it(`should delete only 'renderModule' when there are additional exports`, async () => {
+      tree.create(
+        testTypeScriptFilePath,
+        `
+          import { Path, join } from '@angular-devkit/core';
+          export { renderModule, ServerModule } from '@angular/platform-server';
+        `,
+      );
+
+      const newTree = await schematicRunner.runSchematicAsync(schematicName, {}, tree).toPromise();
+      const content = newTree.readContent(testTypeScriptFilePath);
+      expect(content).toContain(`import { Path, join } from '@angular-devkit/core';`);
+      expect(content).toContain(`export { ServerModule } from '@angular/platform-server';`);
+    });
+
+    it(`should not delete 'renderModule' when it's exported from another module`, async () => {
+      tree.create(
+        testTypeScriptFilePath,
+        `
+          export { renderModule } from '@angular/core';
+        `,
+      );
+
+      const newTree = await schematicRunner.runSchematicAsync(schematicName, {}, tree).toPromise();
+      const content = newTree.readContent(testTypeScriptFilePath);
```

Comment:
nit: `Tree` has `readText` now so `readContent` isn't needed anymore.  Could probably also drop `UnitTestTree` completely in the spec as well.